client_timing_test.go

```diff
@@ -640,3 +640,155 @@ func benchmarkPipelineClient(b *testing.B, parallelism int) {
+}
+
+func BenchmarkClientGetEndToEndWaitConn100Inmemory(b *testing.B) {
+	benchmarkClientGetEndToEndWaitConnInmemory(b, 100)
+}
+
+func BenchmarkClientGetEndToEndWaitConn1000Inmemory(b *testing.B) {
+	benchmarkClientGetEndToEndWaitConnInmemory(b, 1000)
+}
+
+func benchmarkClientGetEndToEndWaitConnInmemory(b *testing.B, parallelism int) {
+	ln := fasthttputil.NewInmemoryListener()
+
+	ch := make(chan struct{})
+	sleepDuration := 50 * time.Millisecond
+	go func() {
+
+		if err := Serve(ln, newFasthttpSleepEchoHandler(sleepDuration)); err != nil {
+			b.Errorf("error when serving requests: %s", err)
+		}
+		close(ch)
+	}()
+
+	c := &Client{
+		MaxConnsPerHost:    1,
+		Dial:               func(addr string) (net.Conn, error) { return ln.Dial() },
+		MaxConnWaitTimeout: 5 * time.Second,
+	}
+
+	requestURI := "/foo/bar?baz=123&sleep=10ms"
+	url := "http://unused.host" + requestURI
+	b.SetParallelism(parallelism)
+	b.RunParallel(func(pb *testing.PB) {
+		var buf []byte
+		for pb.Next() {
+			statusCode, body, err := c.Get(buf, url)
+			if err != nil {
+				if err != ErrNoFreeConns {
+					b.Fatalf("unexpected error: %s", err)
+				}
+			} else {
+				if statusCode != StatusOK {
+					b.Fatalf("unexpected status code: %d. Expecting %d", statusCode, StatusOK)
+				}
+				if string(body) != requestURI {
+					b.Fatalf("unexpected response %q. Expecting %q", body, requestURI)
+				}
+			}
+			buf = body
+		}
+	})
+
+	ln.Close()
+	select {
+	case <-ch:
+	case <-time.After(time.Second):
+		b.Fatalf("server wasn't stopped")
+	}
+}
+
+func newNethttpSleepEchoHandler(sleep time.Duration) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(sleep)
+		w.Header().Set(HeaderContentType, "text/plain")
+		w.Write([]byte(r.RequestURI)) //nolint:errcheck
+	}
+}
+
+func BenchmarkNetHTTPClientGetEndToEndWaitConn1Inmemory(b *testing.B) {
+	benchmarkNetHTTPClientGetEndToEndWaitConnInmemory(b, 1)
+}
+
+func BenchmarkNetHTTPClientGetEndToEndWaitConn10Inmemory(b *testing.B) {
+	benchmarkNetHTTPClientGetEndToEndWaitConnInmemory(b, 10)
+}
+
+func BenchmarkNetHTTPClientGetEndToEndWaitConn100Inmemory(b *testing.B) {
+	benchmarkNetHTTPClientGetEndToEndWaitConnInmemory(b, 100)
+}
+
+func BenchmarkNetHTTPClientGetEndToEndWaitConn1000Inmemory(b *testing.B) {
+	benchmarkNetHTTPClientGetEndToEndWaitConnInmemory(b, 1000)
+}
+
+func benchmarkNetHTTPClientGetEndToEndWaitConnInmemory(b *testing.B, parallelism int) {
+	ln := fasthttputil.NewInmemoryListener()
+
+	ch := make(chan struct{})
+	sleep := 50 * time.Millisecond
+	go func() {
+		if err := http.Serve(ln, newNethttpSleepEchoHandler(sleep)); err != nil && !strings.Contains(
+			err.Error(), "use of closed network connection") {
+			b.Errorf("error when serving requests: %s", err)
+		}
+		close(ch)
+	}()
+
+	c := &http.Client{
+		Transport: &http.Transport{
+			Dial:            func(_, _ string) (net.Conn, error) { return ln.Dial() },
+			MaxConnsPerHost: 1,
```

Comment:
@erikdubbelboer travis got an error of `./client_timing_test.go:757:19: unknown field 'MaxConnsPerHost' in struct literal of type http.Transport` on this line in go version 1.9.x and 1.10.x. What should i do?

Reply:
resloved, using `// +build go1.11`